### PR TITLE
Inject version_slug manually into html_context since RTD no longer does it

### PR DESCRIPTION
**Pull request by astrofrog**
Should hopefully address https://github.com/astropy/astropy/issues/17007 but we'll need to wait until there is a ``latest`` build to know for sure.

Close #17068

### CODE REVIEW

**Comment by astrofrog** [comment]
Yay palindromic issue number! \o/

File: docs/conf.py [hunk 1]
```diff
@@ -321,6 +321,7 @@ def _custom_edit_url(
 # A dictionary of values to pass into the template engine's context for all pages.
 html_context = {
     "default_mode": "light",
+    "version_slug": os.environ.get('READTHEDOCS_VERSION') or '',
```
Comment (pllim) [inline docs/conf.py:324]: Oooo... this looks promising. Won't ever work on local build but I don't think it matters.

Comment (astrofrog) [reply docs/conf.py:324]: Exactly, it only matters to remove the line on latest and stable on RTD

**Review by pllim — APPROVED** [review]
Should we merge then? Only one way to find out!

And if it works, do we need a post release for 6.1.x to fix the search?

**Comment by pllim** [comment]
pre-commit.ci autofix

**Comment by astrofrog** [comment]
Yes to both!

**Comment by pllim** [comment]
Feel free to merge! Ignore devdeps failure (scipy wheels having problems)

**Comment by astrofrog** [comment]
Well for 6.1.x we might as well do a proper new bugfix release instead of a 'post' release, it'll be easy enough.

**Pull request merged**